services/nginz/third_party/nginx-zauth-module/zauth_module.c

```diff
@@ -597,6 +600,22 @@ static ngx_int_t zauth_token_var (ngx_http_request_t * r, ngx_http_variable_valu
         return zauth_set_var(r->pool, v, zauth_token_lookup(t, data));
 }
 
+static ngx_int_t zauth_token_var_user (ngx_http_request_t * r, ngx_http_variable_value_t * v, uintptr_t _) {
+        ZauthToken const * t = ngx_http_get_module_ctx(r, zauth_module);
+        if (t != NULL && zauth_is_authorized_and_allowed(r)) {
```

Comment:
This doesn't look right. First `t` is cast to `ZauthToken` and assumed to be a valid pointer to that struct by `zauth_is_authorized_and_allowed`. Later, it is fetched again and cast to a string pointer.

You might want to store a tagged union in the context, so that this function can discriminate if it's the zauth or oauth case.